src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -35,6 +40,45 @@ private void EmitExpression(BoundExpression expression, bool used)
                 }
             }
 
+            _recursionDepth++;
+
+            if (_recursionDepth > 1)
+            {
+                if (_recursionDepth > Syntax.InternalSyntax.LanguageParser.MaxUncheckedRecursionDepth)
+                {
```

Comment:
@VSadov yeah.  We're now using this logic in 3 places (C# parser, VB parser and here).  I wonder if it makes sense to centralize it.  Perhaps in a later change if it's too disruptive here.